### PR TITLE
WIP:  make corelib no_std

### PR DESCRIPTION
**Pull request by ogoffart**
It doesn't compile with no_std yet, we need to find a solution for thread_local

Also a bunch of dependencies still use the std, such as lyon, weak_table, maybe more

### CODE REVIEW

File: sixtyfps_runtime/corelib/Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,13 @@ path = "lib.rs"
 [features]
 # Allow the viewer to query at runtime information about item types
 rtti = []
+# Expose C ABI
 ffi = []
+# Use the standard library
+std = []
```
Comment (tronical) [inline sixtyfps_runtime/corelib/Cargo.toml:21]: Is it intentional not to enable the std features of euclid, etc. here? (Just curious)

File: sixtyfps_runtime/corelib/lib.rs [hunk 2]
```diff
@@ -21,6 +21,8 @@ only be used with `version = "=x.y.z"` in Cargo.toml.
 */
 #![doc(html_logo_url = "https://sixtyfps.io/resources/logo.drawio.svg")]
 #![deny(unsafe_code)]
+//TODO #![cfg_attr(not(feature = "std"), no_std)]
+extern crate alloc;
```
Comment (tronical) [inline sixtyfps_runtime/corelib/lib.rs:25]: What does the TODO mean?

Comment (ogoffart) [reply sixtyfps_runtime/corelib/lib.rs:25]: That we need to do this:

```suggestion
#![cfg_attr(not(feature = "std"), no_std)]
extern crate alloc;
```

When the crate can compile without the "std" feature :-)

File: sixtyfps_runtime/corelib/Cargo.toml (hunk 1 above)
Comment (ogoffart) [reply sixtyfps_runtime/corelib/Cargo.toml:21]: yes, i don't believe we need them, at least it compiles without it

Comment (tronical) [reply sixtyfps_runtime/corelib/Cargo.toml:21]: That makes sense - great :)

File: sixtyfps_runtime/corelib/lib.rs (hunk 2 above)
Comment (tronical) [reply sixtyfps_runtime/corelib/lib.rs:25]: Ah yes, of course - sorry :)

**Pull request merged**